src/sync/atomic/ptr.rs

```diff
@@ -0,0 +1,123 @@
+use crate::sync::atomic::Atomic;
+use std::sync::atomic::Ordering;
+
+/// A raw pointer type which can be safely shared between threads.
+pub struct AtomicPtr<T> {
+    inner: Atomic<*mut T>,
+}
+
+impl<T> Default for AtomicPtr<T> {
+    fn default() -> Self {
+        Self::new(std::ptr::null_mut())
+    }
+}
+
+impl<T> From<*mut T> for AtomicPtr<T> {
+    fn from(p: *mut T) -> Self {
+        Self::new(p)
+    }
+}
+
+impl<T> std::fmt::Debug for AtomicPtr<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        std::fmt::Debug::fmt(unsafe { &self.raw_load() }, f)
+    }
+}
+
+unsafe impl<T> Send for AtomicPtr<T> {}
```

Comment:
Comments here?

Reply:
This one's actually not a Shuttle-specific hack — it's to indicate that `AtomicPtr` makes a raw pointer Send + Sync even though it's still unsafe to dereference (the std implementation does the same manual impl). Added a comment to that effect.